src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -159,53 +131,51 @@ internal static void IntrospectiveSort(Span<T> keys, Comparison<T> comparer)
 
             if (keys.Length > 1)
             {
-                IntroSort(keys, 2 * IntrospectiveSortUtilities.FloorLog2PlusOne(keys.Length), comparer);
+                IntroSort(keys, 2 * (BitOperations.Log2((uint)keys.Length) + 1), comparer);
             }
         }
 
         private static void IntroSort(Span<T> keys, int depthLimit, Comparison<T> comparer)
         {
-            int lo = 0;
-            int hi = keys.Length - 1;
-
+            Debug.Assert(!keys.IsEmpty);
+            Debug.Assert(depthLimit >= 0);
             Debug.Assert(comparer != null);
 
-            while (hi > lo)
+            int hi = keys.Length - 1;
+            while (hi > 0)
             {
-                int partitionSize = hi - lo + 1;
-                if (partitionSize <= IntrospectiveSortUtilities.IntrosortSizeThreshold)
+                int partitionSize = hi + 1;
+
+                if (partitionSize <= Array.IntrosortSizeThreshold)
                 {
-                    if (partitionSize == 1)
-                    {
-                        return;
-                    }
+                    Debug.Assert(partitionSize >= 2);
 
                     if (partitionSize == 2)
                     {
-                        SwapIfGreater(keys, comparer, lo, hi);
+                        SwapIfGreater(keys, comparer, 0, hi);
                         return;
                     }
 
                     if (partitionSize == 3)
                     {
-                        SwapIfGreater(keys, comparer, lo, hi - 1);
-                        SwapIfGreater(keys, comparer, lo, hi);
+                        SwapIfGreater(keys, comparer, 0, hi - 1);
+                        SwapIfGreater(keys, comparer, 0, hi);
                         SwapIfGreater(keys, comparer, hi - 1, hi);
                         return;
                     }
 
-                    InsertionSort(keys[lo..(hi+1)], comparer);
+                    InsertionSort(keys.Slice(0, hi+1), comparer);
```

Comment:
Have you changed this from `[x..y]` to `Slice` for cosmetic reasons, or for performance reasons? (just wondering)

Reply:
Perf:
https://github.com/dotnet/roslyn/issues/43598